### PR TITLE
Create and remove -Milestone 2

### PR DESCRIPTION
**Pull request by ahmetbozaci**
In this milestone;

- Create `book reducer, actions, action types` with using `duck pattern`.
- Configure store.
- Create `add` and `remove` function.

_I will update style end of the project._

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+import store from './redux/configureStore';
 
-function App() {
-  return <Navbar />;
-}
+const App = () => (
+  <Provider store={store}>
+    <Navbar />
+  </Provider>
+);
```
Comment (omar-labana) [inline src/App.js]: - [x] The `<Navbar />` component should be responsible only for navigation, kindly create a new component separating the `<Navbar />` and the main home page of your project then update your components routing.
Should be something like this:
```jsx
<Provider store={store}>
  <Navbar />
  <Router>
    <Switch>
      <Route path="/books">
        <Books />
      </Route>
      <Route path="/categories">
        <Categories />
      </Route>
    </Switch>
  </Router>
</Provider>
```
- [x] Please update your routes to route the visitor directly to the books page as the `path='/'`, currently, I have to click on the `Books` link to get there.

File: src/components/book/BookForm.jsx [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
+    const author = document.getElementById('author').value;
+    const newBook = {
+      id: uuidv4(),
+      title,
+      author,
+    };
+    dispatch(addBook(newBook));
+    document.getElementById('title').value = '';
+    document.getElementById('author').value = '';
+  };
```
Comment (omar-labana) [inline src/components/book/BookForm.jsx]: - [x] Kindly, update this component. It should contain a local state with `title` and `author`. The state should be updated using the (`onChange`) listener on each input field.
- [x] Submitting an item to the store should take the input from the component's local state, then invoke `dispatch(addBook(newBook))`.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @ahmetbozaci, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_N/A_
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/book/BookForm.jsx (hunk 2 above)
Comment (ahmetbozaci) [reply src/components/book/BookForm.jsx]: Hello @omar-labana,
I don't understand this part. Local state should empty. I had it then I delete it after see requirement. Also why I need `onChange` listener. This is working with `addBookStore` function. Function  is adding to store correctly.

Comment (omar-labana) [reply src/components/book/BookForm.jsx]: Hi @ahmetbozaci,
You should save any input data by the user in a local state using `useState` react hook, it's the standard for second-tier form validation that you may actually use later, it's also one of the requirements of the project.
If you need more info, kindly reach out on slack.

**Review by dasileker — APPROVED** [review]
## **Hi @ahmetbozaci,** 

### **Nice Job man 👌**

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 
Approved


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**